cli/src/semgrep/join_rule.py

```diff
@@ -369,18 +369,47 @@ def cli_intermediate_vars_to_core_intermediate_vars(
     return [core.CoreMatchIntermediateVar(location=v.location) for v in i_vars]
 
 
+def cli_call_trace_to_core_call_trace(
+    trace: core.CliMatchCallTrace,
+) -> core.CoreMatchCallTrace:
+    value = trace.value
+    if isinstance(value, core.CliLoc):
+        return core.CoreMatchCallTrace(core.CoreLoc(value.value[0]))
+    elif isinstance(value, core.CliCall):
+        data, intermediate_vars, call_trace = value.value
+
+        return core.CoreMatchCallTrace(
+            core.CoreCall(
+                (
+                    data[0],
+                    cli_intermediate_vars_to_core_intermediate_vars(intermediate_vars),
```

Comment:
For my own understanding, why do we convert the cli vars back to core vars?

Reply:
pretty much just because the historic join rule code uses core stuff, whereas it receives cli stuff. no idea why it's necessary, but i just made the types fit